server.py

```diff
@@ -141,68 +204,211 @@ def connect_client(self):
 
         self.client_number[client_number] = FREE
         self.connections_needed_to_start += CLOSED_CONNECTION
         self.number_of_started_connections -= CLOSED_CONNECTION
         self.number_of_started_games -= GAME_IS_ENDED
         self.client_sockets[client_number].close()
+        self.chosen_champions[client_number] = CHAMPION_IS_NOT_CHOSEN
         sys.exit()
 
     def champion_select(self, client_number):
-        chosen_champion = pickle.loads(self.client_sockets[client_number].recv(BUFFER_SIZE))
+        try:
+            chosen_champion = pickle.loads(self.client_sockets[client_number].recv(BUFFER_SIZE))
 
-        if chosen_champion == CLIENT_QUIT:
-            return LEFT_CHAMPION_SELECT
+            if chosen_champion == CLIENT_QUIT:
+                return LEFT_CHAMPION_SELECT
 
-        if self.server_status == SERVER_IS_CLOSING:
-            self.client_sockets[client_number].sendall(SERVER_QUIT_SIGNAL_IN_CHAMPION_SELECT)
-            return LEFT_CHAMPION_SELECT
+            if self.server_status == SERVER_IS_CLOSING:
+                self.client_sockets[client_number].sendall(SERVER_QUIT_SIGNAL_IN_CHAMPION_SELECT)
+                return LEFT_CHAMPION_SELECT
 
-        if chosen_champion in self.chosen_champions:
-            self.client_sockets[client_number].sendall(CHAMPION_IS_NOT_AVAILABLE)
-            return CHAMPION_SELECT
+            if chosen_champion in self.chosen_champions:
+                self.client_sockets[client_number].sendall(CHAMPION_IS_NOT_AVAILABLE)
+                return CHAMPION_SELECT
 
-        else:
-            self.client_sockets[client_number].sendall(CHAMPION_IS_AVAILABLE)
-            self.chosen_champions[client_number] = chosen_champion
-            return LOBBY
+            else:
+                self.client_sockets[client_number].sendall(CHAMPION_IS_AVAILABLE)
+                self.chosen_champions[client_number] = chosen_champion
+                return LOBBY
+        except (ConnectionResetError, ConnectionAbortedError):
+            return LEFT_CHAMPION_SELECT
 
     def lobby(self, client_number):
-        client_signal = self.client_sockets[client_number].recv(BUFFER_SIZE).decode()
+        try:
+            client_signal = self.client_sockets[client_number].recv(BUFFER_SIZE).decode()
 
-        if self.server_status == SERVER_IS_CLOSING:
-            self.client_sockets[client_number].send(pickle.dumps(SERVER_QUIT_SIGNAL_IN_LOBBY))
+            if self.server_status == SERVER_IS_CLOSING:
+                self.client_sockets[client_number].send(pickle.dumps(SERVER_QUIT_SIGNAL_IN_LOBBY))
+                self.client_sockets[client_number].send(pickle.dumps(SERVER_QUIT_SIGNAL_IN_LOBBY))
+                return CONNECTION_IS_DEACTIVATED
+
+            if client_signal == CHOSEN_CHAMPIONS_INFORMATION_REQUEST:
+                self.client_sockets[client_number].send(pickle.dumps(self.chosen_champions))
+
+                count = EVERY_PLAYER_CONNECTED
```

Comment:
instead of count I would add active_players or players_in_lobby

Reply:
changed